lib/Dialect/Moore/Transforms/SimplifyProcedures.cpp

```diff
@@ -53,7 +53,8 @@ void SimplifyProceduresPass::runOnOperation() {
         // Collect the users of the global variable that is mentioned above.
         DenseSet<Operation *> users;
         for (auto *user : nestedOp.getOperand(0).getUsers())
-          if (!users.contains(user))
+          // Ensuring don't handle the users existing in another procedure body.
+          if (user->getBlock() == procedureOp.getBody())
             users.insert(user);
```

Comment:
I learned that the underlying implementation of the `insert()` of `llvm::DenseSet` is through calling the `try_emplace` of `llvm::DensMap`. So I think we needn't estimate whether the user exists in the `DenseSet`.